plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/yq/yq.ts

```diff
@@ -0,0 +1,130 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createTemplateAction } from '@backstage/plugin-scaffolder-backend';
+import { resolveSafeChildPath } from '@backstage/backend-common';
+import { execFile, ChildProcess } from 'child_process';
+import { promisify } from 'util';
+import fs from 'fs';
+
+const execFilePromise = promisify(execFile);
+
+const createMessageSanitizer = (ctx: { workspacePath: string }) => {
+  return (message: string) => {
+    return message.replace(ctx.workspacePath, '<redacted>');
+  };
+};
+
+const wait = async (
+  process: ChildProcess,
+): Promise<{ stdout: string; stderr: string }> => {
+  return await new Promise((resolve, reject) => {
+    let stdout = '';
+    let stderr = '';
+    process.stdout?.on('data', data => {
+      stdout = stdout.concat(data);
+    });
+    process.stderr?.on('data', data => {
+      stderr = stderr.concat(data);
+    });
+    process.on('exit', code => {
+      if (code !== 0) {
+        reject(new Error(`process failed: ${stderr}`));
+      }
+      resolve({ stdout, stderr });
+    });
+  });
+};
+
+export function createYqAction({ actionId }: { actionId?: string }) {
+  return createTemplateAction<{
+    path: string;
+    query: string[] | string;
+    writeOutputPath?: string;
+  }>({
+    id: actionId || 'roadiehq:utils:yq',
+    description: 'Allows using yq syntax on a file in the workspace.',
+    schema: {
+      input: {
+        type: 'object',
+        required: ['query', 'path'],
+        properties: {
+          path: {
+            title: 'Path',
+            description: 'Path read the file from.',
+            type: 'string',
+          },
+          query: {
+            description: 'Array of query items to run in jq.',
+            title: 'Query',
+            type: ['array', 'string'],
+          },
+          writeOutputPath: {
+            description: 'Write the output of the query back to the file.',
+            title: 'Write Output path',
+            type: 'string',
+          },
+        },
+      },
+      output: {
+        type: 'object',
+        properties: {
+          result: {
+            title: 'Output',
+            type: 'string',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      const messageSanitizer = createMessageSanitizer(ctx);
+      const sourceFilepath = resolveSafeChildPath(
+        ctx.workspacePath,
+        ctx.input.path,
+      );
+      const queries = [ctx.input.query].flat();
+
+      ctx.logger.info(`Attempting to yq with ${ctx.input.path}`);
+
+      const writePath =
+        ctx.input.writeOutputPath &&
+        resolveSafeChildPath(ctx.workspacePath, ctx.input.writeOutputPath);
+      if (!fs.existsSync(sourceFilepath)) {
+        throw new Error(`${messageSanitizer(sourceFilepath)} does not exist`);
+      }
+      try {
+        const initialResult = await execFilePromise('yq', [sourceFilepath]);
```

Comment:
This type of code that depends on a cli command is very hard to test.

Reply:
Does it have to be a cli command?

Reply:
I havent found a library for this yet. Suggestions welcome.

Reply:
`yaml` on NPM would allow you to achieve this

Reply:
I think its the merging nested arrays part that Ive trouble with. How would that be surfaced via an action?

e.g.

if I have a yaml file containing:

```yaml
person:
   numbers:
    - 555123456
    - 555123457
```

How do I form an action to append a new number to that list and write it back to the file?

With the action I have just created I can do that with:

```yaml
    - id: yamlmerge
      name: YAML Merge
      action: roadiehq:utils:yq
      input:
        path: person.yaml
        writeOutputPath: person.yaml
        query:
          - '(.person.numbers = .person.numbers + 555123458)'
```

Reply:
It also allows for deleting that phone number later.